components/eamxx/src/diagnostics/tests/number_paths_tests.cpp

```diff
@@ -0,0 +1,243 @@
+    const auto &ni_f          = input_fields.at("ni");
+    const auto &ni_v          = ni_f.get_view<Real **>();
+    const auto &qr_f          = input_fields.at("qr");
+    const auto &qr_v          = qr_f.get_view<Real **>();
+    const auto &nr_f          = input_fields.at("nr");
+    const auto &nr_v          = nr_f.get_view<Real **>();
+    const auto &pseudo_dens_f = input_fields.at("pseudo_density");
+    const auto &pseudo_dens_v = pseudo_dens_f.get_view<Real **>();
+    for(int icol = 0; icol < ncols; icol++) {
+      const auto &qc_sub = ekat::subview(qc_v, icol);
+      const auto &nc_sub = ekat::subview(nc_v, icol);
+      const auto &qi_sub = ekat::subview(qi_v, icol);
+      const auto &ni_sub = ekat::subview(ni_v, icol);
+      const auto &qr_sub = ekat::subview(qr_v, icol);
+      const auto &nr_sub = ekat::subview(nr_v, icol);
+      const auto &dp_sub = ekat::subview(pseudo_dens_v, icol);
+      ekat::genRandArray(pseudo_density, engine, pdf_pseudodens);
+      Kokkos::deep_copy(dp_sub, pseudo_density);
+
+      ekat::genRandArray(qc, engine, pdf_qx);
+      ekat::genRandArray(nc, engine, pdf_qx);
+      ekat::genRandArray(qi, engine, pdf_qx);
+      ekat::genRandArray(ni, engine, pdf_qx);
+      ekat::genRandArray(qr, engine, pdf_qx);
+      ekat::genRandArray(nr, engine, pdf_qx);
+      Kokkos::deep_copy(qc_sub, qc);
+      Kokkos::deep_copy(nc_sub, nc);
+      Kokkos::deep_copy(qi_sub, qi);
+      Kokkos::deep_copy(ni_sub, ni);
+      Kokkos::deep_copy(qr_sub, qr);
+      Kokkos::deep_copy(nr_sub, nr);
+    }
+    // Grab views for each of the number path diagnostics
+    const auto &lnp   = diags["lnp"]->get_diagnostic();
+    const auto &inp   = diags["inp"]->get_diagnostic();
+    const auto &rnp   = diags["rnp"]->get_diagnostic();
+    const auto &lnp_v = lnp.get_view<const Real *>();
+    const auto &inp_v = inp.get_view<const Real *>();
+    const auto &rnp_v = rnp.get_view<const Real *>();
+    const auto &lnp_h = lnp.get_view<const Real *, Host>();
+    const auto &inp_h = inp.get_view<const Real *, Host>();
+    const auto &rnp_h = rnp.get_view<const Real *, Host>();
+
+    for(const auto &dd : diags) {
+      dd.second->compute_diagnostic();
+    }
+    // test manual calculation vs one provided by diags
+    {
+      Kokkos::parallel_for(
+          "", policy, KOKKOS_LAMBDA(const MemberType &team) {
+            const int icol = team.league_rank();
+            auto dp_icol   = ekat::subview(pseudo_dens_v, icol);
+            // Liquid
+            Real qndc_prod = 0.0;
+            auto qc_icol   = ekat::subview(qc_v, icol);
+            auto nc_icol   = ekat::subview(nc_v, icol);
+            Kokkos::parallel_reduce(
+                Kokkos::TeamVectorRange(team, num_levs),
+                [&](Int idx, Real &lsum) {
+                  lsum += nc_icol(idx) * qc_icol(idx) * dp_icol(idx) / gravit;
+                },
+                qndc_prod);
+            REQUIRE(std::abs(lnp_v(icol) - qndc_prod) < macheps);
```

Comment:
REQUIRE is a Catch2 macro, that does not work on device. You have two options:
- turn `parallel_for` into `parallel_reduce`, where you reduce a counter of how many failed comparisons you get.
- just do a simple for loop without kokkos parallel stuff (and compare against host views from the diags).

Option two would be my favorite, but you may run into issues with order of ops in the reduction.